app/components/post_list/load_more_posts.js

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent, PropTypes} from 'react';
+import {
+    View,
+    StyleSheet,
+    TouchableOpacity
+} from 'react-native';
+
+import {makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import FormattedText from 'app/components/formatted_text';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return StyleSheet.create({
+        container: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            justifyContent: 'center',
+            height: 28,
+            marginVertical: 10
+        },
+        text: {
+            fontSize: 14,
+            fontWeight: '600',
+            color: theme.linkColor
+        }
+    });
+});
+
+export default class LoadMorePosts extends PureComponent {
+    static propTypes = {
+        loading: PropTypes.bool.isRequired,
+        loadMore: PropTypes.func,
+        theme: PropTypes.object.isRequired,
+        style: View.propTypes.style
+    };
+
+    loadMore = () => {
+        const {loading, loadMore} = this.props;
+        if (!loading && typeof loadMore === 'function') {
+            loadMore();
+        }
+    };
+
+    renderText(style) {
+        let [id, defaultMessage] = ['posts_view.loadMore', 'Load more messages'];
```

Comment:
@hmhealey Yes the destructuring is unnecessary here, but I think it helps clarify what's going on. `id` and `defaultMessage` are entirely coupled. You wouldn't override one and not the other. Declaring them and overriding them both in one line helps emphasize the coupling. Plus I would've felt the need to come up with a more descriptive variable than `id` if it was declared separately from `defaultMessage`.